### PR TITLE
Provide option to unlink a fork

### LINKED ISSUE #4566: Make fork independent from original repository
- Gitea version (or commit ref): 1.4.3
- Git version: 
- Operating system:
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

A fork currently can not be converted to a "normal" repository other than by pushing into a new repository. When a repository is a fork, pull requests will be pointing to the original repository by default, which might not be wanted in case the fork diverges.

### PR DESCRIPTION
**Pull request by zeripath**
Provides a setting option to unlink a fork from its base repository. 

Fix #4566

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -1372,6 +1372,11 @@ settings.convert_desc = You can convert this mirror into a regular repository. T
 settings.convert_notices_1 = This operation will convert the mirror into a regular repository and cannot be undone.
 settings.convert_confirm = Convert Repository
 settings.convert_succeed = The mirror has been converted into a regular repository.
+settings.convert_fork = Convert to Regular Repository
+settings.convert_fork_desc = You can convert this fork into a regular repository. This cannot be undone.
+settings.convert_fork_notices_1 = This operation will convert the mirror into a regular repository and cannot be undone.
+settings.convert_fork_confirm = Convert Repository
+settings.convert_fork_succeed = The mirror has been converted into a regular repository.
```
Comment (techknowlogick) [inline options/locale/locale_en-US.ini]:
```suggestion
settings.convert_fork_succeed = The fork has been converted into a regular repository.
```

File: options/locale/locale_en-US.ini [hunk 2]
```diff
@@ -1372,6 +1372,11 @@ settings.convert_desc = You can convert this mirror into a regular repository. T
 settings.convert_notices_1 = This operation will convert the mirror into a regular repository and cannot be undone.
 settings.convert_confirm = Convert Repository
 settings.convert_succeed = The mirror has been converted into a regular repository.
+settings.convert_fork = Convert to Regular Repository
+settings.convert_fork_desc = You can convert this fork into a regular repository. This cannot be undone.
+settings.convert_fork_notices_1 = This operation will convert the mirror into a regular repository and cannot be undone.
```
Comment (techknowlogick) [inline options/locale/locale_en-US.ini]:
```suggestion
settings.convert_fork_notices_1 = This operation will convert the fork into a regular repository and cannot be undone.
```

**Comment by CirnoT** [comment]
I don't think we should push it before we have a way to limit forks user can create, otherwise all instances will suffer from repository limit being broken

**Comment by zeripath** [comment]
We already have a convert mirror which I think also get round the repo counts too. I had forgotten about the repo counts - I guess we can prevent the conversion if they're at the limit.

File: models/repo.go [hunk 3]
```diff
@@ -208,19 +208,9 @@ func (repo *Repository) SanitizedOriginalURL() string {
...
-	} else if repo.Owner != nil {
-		ownerName = repo.Owner.Name
-	} else {
-		ownerName = log.NewColoredIDValue(strconv.FormatInt(repo.OwnerID, 10))
-	}
-
 	log.ColorFprintf(s, "%d:%s/%s",
 		log.NewColoredIDValue(repo.ID),
-		ownerName,
+		repo.OwnerName,
```
Comment (zeripath) [inline models/repo.go:214]: This set of changes just tidies up this function since we changed to always store the OwnerName

**Comment by zeripath** [comment]
@CirnoT This will now check if you can create repos before transferring.

**Comment by CirnoT** [comment]
That should be fine then

**Comment by CirnoT** [comment]
```diff
Index: web_src/less/_repository.less
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- web_src/less/_repository.less	(revision cce3f9f6fccbcd4dbcadc28c605ef334e74f9baf)
+++ web_src/less/_repository.less	(date 1592203778992)
@@ -2744,7 +2744,10 @@
 }
 
 #transfer-repo-modal,
-#delete-repo-modal {
+#delete-repo-modal,
+#delete-wiki-modal,
+#convert-fork-repo-modal,
+#convert-mirror-repo-modal {
     .ui.message {
         width: 100% !important;
     }
Index: templates/repo/settings/options.tmpl
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- templates/repo/settings/options.tmpl	(revision cce3f9f6fccbcd4dbcadc28c605ef334e74f9baf)
+++ templates/repo/settings/options.tmpl	(date 1592203778999)
@@ -360,7 +360,7 @@
 			{{if .Repository.IsMirror}}
 				<div class="item">
 					<div class="ui right">
-						<button class="ui basic red show-modal button" data-modal="#convert-repo-modal">{{.i18n.Tr "repo.settings.convert"}}</button>
+						<button class="ui basic red show-modal button" data-modal="#convert-mirror-repo-modal">{{.i18n.Tr "repo.settings.convert"}}</button>
 					</div>
 					<div>
 						<h5>{{.i18n.Tr "repo.settings.convert"}}</h5>
@@ -448,7 +448,7 @@
 
 {{if .Permission.IsOwner}}
 	{{if .Repository.IsMirror}}
-		<div class="ui small modal" id="convert-repo-modal">
+		<div class="ui small modal" id="convert-mirror-repo-modal">
 			<div class="header">
 				{{.i18n.Tr "repo.settings.convert"}}
 			</div>
```

**Comment by CirnoT** [comment]
Seems good otherwise

**Review by CirnoT — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
make lg-tm work

**Pull request merged**